src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,244 @@
+                SyntaxKind.LessThanExpression,
+                SyntaxKind.LessThanOrEqualExpression,
+                SyntaxKind.GreaterThanExpression,
+                SyntaxKind.GreaterThanOrEqualExpression);
+
+        private static readonly ImmutableArray<SyntaxKind> s_syntaxKindsNeedsToCheckThePrecedence =
+            ImmutableArray.Create(
+                SyntaxKind.AddExpression,
+                SyntaxKind.SubtractExpression,
+                SyntaxKind.MultiplyExpression,
+                SyntaxKind.DivideExpression,
+                SyntaxKind.ModuloExpression,
+                SyntaxKind.LeftShiftExpression,
+                SyntaxKind.RightShiftExpression,
+                SyntaxKind.LogicalOrExpression,
+                SyntaxKind.LogicalAndExpression,
+                SyntaxKind.BitwiseOrExpression,
+                SyntaxKind.BitwiseAndExpression,
+                SyntaxKind.ExclusiveOrExpression,
+                SyntaxKind.EqualsExpression,
+                SyntaxKind.NotEqualsExpression,
+                SyntaxKind.LessThanExpression,
+                SyntaxKind.LessThanOrEqualExpression,
+                SyntaxKind.GreaterThanExpression,
+                SyntaxKind.GreaterThanOrEqualExpression,
+                SyntaxKind.IsExpression,
+                SyntaxKind.UnaryMinusExpression,
+                SyntaxKind.UnaryPlusExpression,
+                SyntaxKind.LogicalNotExpression,
+                SyntaxKind.BitwiseNotExpression,
+                SyntaxKind.CastExpression,
+                SyntaxKind.IsPatternExpression,
+                SyntaxKind.AsExpression,
+                SyntaxKind.CoalesceExpression,
+                SyntaxKind.AwaitExpression,
+                SyntaxKind.ConditionalAccessExpression,
+                SyntaxKind.SimpleMemberAccessExpression,
+                SyntaxKind.ParenthesizedLambdaExpression,
+                SyntaxKind.InvocationExpression,
+                SyntaxKind.SimpleLambdaExpression,
+                SyntaxKind.ElementAccessExpression,
+                SyntaxKind.SwitchExpression,
+                SyntaxKind.ConditionalExpression,
+                SyntaxKind.SuppressNullableWarningExpression,
+                SyntaxKind.RangeExpression,
+                SyntaxKind.SimpleAssignmentExpression,
+                SyntaxKind.AddAssignmentExpression,
+                SyntaxKind.SubtractAssignmentExpression,
+                SyntaxKind.MultiplyAssignmentExpression,
+                SyntaxKind.DivideAssignmentExpression,
+                SyntaxKind.ModuloAssignmentExpression,
+                SyntaxKind.AndAssignmentExpression,
+                SyntaxKind.OrAssignmentExpression,
+                SyntaxKind.ExclusiveOrAssignmentExpression,
+                SyntaxKind.RightShiftAssignmentExpression,
+                SyntaxKind.LeftShiftAssignmentExpression
+            );
+
+        private static readonly ImmutableArray<SyntaxKind> s_syntaxKindsConsideredAsStatementInvokesCallee =
```

Comment:
provide comments on these three static fields.

Reply:
Actually it is used to figure out if parenthesis is need or not. Since there is a better way so they can be removed;)